bitshuffle.py

```diff
@@ -133,103 +133,96 @@ def encode_file(fhandle, chunksize, compresslevel, compresstype, filename):
+    parser.add_argument("--compresslevel", '-m', type=int,
                         help="Compression level when encoding. " +
-                        "1 is lowest, 9 is highest")
+                        "1 is lowest, 9 is highest. Defaults to 5.")
 
     parser.add_argument("--editor", "-E",
-                        help="Editor to use for pasting packets")
-
-    parser.add_argument("--compresstype", '-t', default="bz2",
-                        help="Type of compression to use. Defaults to bz2. " +
-                             "Ignored if decoding packets. " +
+                        help="Editor to use for pasting packets. " +
+                        "If not specified, defaults in this order:\n" +
+                        "\t$VISUAL, $EDITOR, mimeopen, nano, vi, " +
+                        "emacs, micro, notepad, notepad++")
+
+    parser.add_argument("--compresstype", '-t',
+                        help="Type of compression to use when encoding. "
+                             "Defaults to bz2. " +
                              "Currently supported: 'bz2', 'gzip'")
 
     args = parser.parse_args()
 
-    if args.version:
-        print("Version: bitshuffle v{0}".format(program_version))
+    # Checks if no parameters were passed
+    if len(sys.argv[1:]) == 0:
+        parser.print_usage()
+        sys.exit(1)
+
+    elif args.version:
+        print("Version: bitshuffle v%s" % version)
         exit(0)
 
-    if args.filename is None:
-        args.filename = os.path.basename(args.input)
+    # Encode & Decode inference
+    args = infer_mode(args)
+
+    # Set default values
+    args = set_defaults(args)
 
+    # Main
     if args.encode:
+
         if args.compresstype not in ['bz2', 'gzip']:
             parser.print_help()
             sys.exit(1)
 
-        with open(args.input, 'rb') as f:
-            packets = encode_file(f, args.chunksize, args.compresslevel,
-                                  args.compresstype, args.filename)
-            with open(args.output, 'w') as of:
-                for p in packets:
-                    of.write(p)
-                    of.write("\n\n")
+        if check_for_file(args.input):
+            with open(args.input, 'rb') as f:
+                packets = encode_file(f, args.chunksize, args.compresslevel,
+                                      args.compresstype, args.filename)
+                with open(args.output, 'w') as of:
+                    for p in packets:
+                        of.write(p)
+                        of.write("\n\n")
 
-                of.flush()
+                    of.flush()
+        else:
+            quit('Error: Input file not found')
 
     elif args.decode:
+
         infile = args.input
         # set to True for infile to be deleted after decoding
         is_tmp = False
-        if stdin.isatty() and args.input is '/dev/stdin':
+        if stdin.isatty() and args.input == '/dev/stdin':
```

Comment:
As an aside, we should find a portable alternative to `/dev/stdin`, as Windows does not support it, and it's not a POSIX thing to my knowledge. On old school UNIX `/dev/stdin` comes from fdescfs, and from procfs on Linux, but it might not be present on all UNIX variants. Python probably has libraries for dealing with this. Not something that needs to be fixed right now, but should probably be added as an issue.